traveler-ios-ui/TravelerKitUI/ViewControllers/OrderDetailViewController.swift

```diff
@@ -53,10 +81,41 @@ class OrderDetailViewController: UITableViewController {
     }
 
     @IBAction func didCancelOrder(_ sender: Any) {
-        
+        if order?.status != OrderStatus.cancelled {
```

Comment:
this kind of condition checking should be done in a guard statement.

Though here it is unnecessary. The button should be disabled if the order is cancelled anyway, it will never be run

Reply:
unresolved

Reply:
Condition is now a guard statement. However, I don't think it is unnecessary, according to UX requirements, if order is cancelled then button will display order receipt. I instead added a switch statement to keep track of the state.